### PR TITLE
Added PassengerDeletion.Anim

### PR DESCRIPTION
**Pull request by FS-21**
Added PassengerDeletion.Anim tag. 
Is an optional animation when the passenger is erased.

### CODE REVIEW

File: docs/New-or-Enhanced-Logics.md [hunk 1]
```diff
@@ -268,11 +268,12 @@ DefaultDisguise=E2  ; InfantryType
 In `rulesmd.ini`:
 ```ini
 [SOMETECHNO]                            ; TechnoType
-PassengerDeletion.Rate=                 ; integer, game frames
+PassengerDeletion.Rate=0                 ; integer, game frames
```
Comment (Metadorius) [inline docs/New-or-Enhanced-Logics.md:271]:
```suggestion
PassengerDeletion.Rate=0                ; integer, game frames
```

File: src/Ext/Techno/Body.cpp [hunk 2]
```diff
@@ -358,6 +358,16 @@ void TechnoExt::EatPassengers(TechnoClass* pThis)
...
 
+							const auto pAnimType = pData->PassengerDeletion_Anim.Get(nullptr);
+							if (pAnimType)
+							{
+								if (auto const pAnim = GameCreate<AnimClass>(pAnimType, pThis->Location))
+								{
+									pAnim->SetOwnerObject(pThis);
+									pAnim->Owner = pThis->Owner;
+								}
+							}
```
Comment (Metadorius) [inline src/Ext/Techno/Body.cpp:369]: Better check if the value was set with `isset`.

**Review by Metadorius — CHANGES_REQUESTED** [review]

**Comment by FS-21** [comment]
Applied suggested review changes.

**Review by Starkku — APPROVED** [review]
Feature seems to work as expected and code looks fine. Approving & merging.

**Pull request merged**